lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -239,10 +282,22 @@ def error_code_from(action, response)
         code&.to_s
       end
 
+      def error_code_from_get(response)
+        return if success_from_get(response)
+        if !response.instance_of? Array
+          code = response['status_code'] || response['code']
+          code&.to_s
+        end
```

Comment:
Same here. Looks like we are checking for just this scenario.

Reply:
Addressed. Added more conditions to handle all type of responses.